client/src/main/java/org/apache/celeborn/client/read/ChunkClient.java

```diff
@@ -246,17 +246,17 @@ private class RetryingChunkReceiveCallback implements ChunkReceivedCallback {
     }
 
     @Override
-    public void onSuccess(int chunkIndex, ManagedBuffer buffer) {
-      callback.onSuccess(chunkIndex, buffer);
+    public void onSuccess(int chunkIndex, ManagedBuffer buffer, PartitionLocation location) {
+      callback.onSuccess(chunkIndex, buffer, ChunkClient.this.location);
```

Comment:
It's wired that when RetryingChunkReceiveCallback.onSuccess is called in TransportResponseHandler, null is passed as location. I think it's not a good idea to make RetryingChunkReceiveCallback a subclass of ChunkReceivedCallback, just make callback and location a class member of RetryingChunkReceiveCallback. TransportResponseHandler's outstandingFetches be type of RetryingChunkReceiveCallback, and make RetryingChunkReceiveCallback public.

Reply:
We can do this in later pr